src/cbmc/bmc.cpp

```diff
@@ -588,6 +589,11 @@ safety_checkert::resultt bmct::stop_on_fail(
   const goto_functionst &goto_functions,
   prop_convt &prop_conv)
 {
+  if(options.get_bool_option("localize-faults"))
+  {
+    //ENHANCE: currently, requires only freezing of guards
+    prop_conv.set_all_frozen();
+  }
```

Comment:
I would probably do the solving in fault_localizationt, and then do the freezing of the guards before the solving